### PR TITLE
Update and build with OpenSSL 3 [skip ci]

### PR DESCRIPTION
**Pull request by JeanChristopheMorinPerso**
Update to 1.3.18 (which s safe to do based on https://abi-laboratory.pro/tracker/timeline/freetds/ and build with OpenSSL 3.

Also some cleanup.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-{% set sha256 = "dba02e7f0661ff42dc289fc41d7cde7e03089fd326f08ee28c872ac9ff4a1c84" %}
+{% set version = "1.3.18" %}
+{% set sha256 = "1d8561d57c71991a28f4681343785c23a6a3eb54d5bcd23897d07e3825ff2d56" %}
 
 package:
   name: {{ name }}
   version: {{ version }}
 
 source:
   fn: freetds-{{ version }}.tar.gz
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
```
This can be removed as it's not used anymore

**Review by skupr-anaconda — APPROVED** [review]
LGTM

File: recipe/meta.yaml [hunk 2]
```diff
@@ -47,13 +47,14 @@ test:
     - conda inspect objects freetds  # [osx]
 
 about:
-  home: http://www.freetds.org/
-  license: LGPL-2.0 AND GPL-2.0
-  license_file: 
+  home: https://www.freetds.org/
+  license: LGPL-2.0
+  license_file:
     - COPYING_LIB.txt
```
Comment (boldorider4) [inline recipe/meta.yaml:53]: why do we not need to include both COPYING.txt and COPYING_LIB.txt?

Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml:53]: `COPYING_LIB.txt` contains the text of the LGPLv2 license while `COPYING.txt` is GPLv2. The project page states that they use an LGPLv2 license.

* https://github.com/FreeTDS/freetds#readme-for-freetds-14dev-development
* https://www.freetds.org/faq.html#license

**Review by boldorider4 — APPROVED** [review]

**Pull request merged**